src/modules/common/helpers/mappings.ts

```diff
@@ -0,0 +1,73 @@
+import { IHomeCarouselData } from "../../home/types/home-types";
+import useMangaStore from "../stores/store";
+import { ITrendingMangaApp } from "../types/app-types";
+
+export const getViewAllCategoryHeading = (category: string) => {
+    switch (category) {
+        case "trending":
+            return "Trending Mangas"
+            break;
+        case "new":
+            return "Latest Mangas"
+            break;
+        default:
+            break;
+    }
+}
+
+export const getCarouselDetails = (category: string): IHomeCarouselData => {
+    switch (category) {
+        case 'new':
+            {
+                return { heading: "Fresh off The Pen", queryKey: "fetchNewMangas" }
+            }
+            break;
+        case 'rank':
+            {
+                return { heading: "Chart-Topping Manga Marvels", queryKey: "fetchTopRankedMangas" }
+            }
+            break;
+        case 'popular':
+            {
+                return { heading: "Tales that Hooked the Hoard", queryKey: "fetchMostFollowedMangas" }
```

Comment:
Nice play on words for the headings. For this one specifically, it might be a bit hard for someone who doesn't speak English as well to understand what it means. So, I'd suggest maybe adding "Popular" somewhere in the heading or around the elements. So like "Popular tales that hooked the hoard"...